indexer/feature_impl.cpp

```diff
@@ -65,12 +65,12 @@ bool IsHouseNumberDeepCheck(strings::UniString const & s)
 
 bool IsHouseNumber(string const & s)
 {
-  return (!s.empty() && IsDigit(s[0]));
+  return (!s.empty() && (IsDigit(s.front()) || IsDigit(s.back())));
 }
 
 bool IsHouseNumber(strings::UniString const & s)
 {
-  return (!s.empty() && IsDigit(s[0]));
+  return (!s.empty() && (IsDigit(s.front()) || IsDigit(s.back())));
```

Comment:
It’s an incremental improvement for editor and search: more house numbers will be recognized and stored correctly in feature’s house number field.

I can improve this code by checking for at least one digit in the string, what do you think about it?

> On Apr 8, 2016, at 12:52, ygorshenin <notifications@github.com> wrote:
> 
> In indexer/feature_impl.cpp <https://github.com/mapsme/omim/pull/2842#discussion_r59003263>:
> 
> >  }
> >  
> >  bool IsHouseNumber(strings::UniString const & s)
> >  {
> > -  return (!s.empty() && IsDigit(s[0]));
> > +  return (!s.empty() && (IsDigit(s.front()) || IsDigit(s.back())));
> Same here, I agree with @mpimenov <https://github.com/mpimenov> that the condition is weaker now, but why do we need it? It won't work for "владение 1 корпус А". Agree that this never worked, but we should do this smarter. What problem this code tries to solve?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2842/files/adf7724a884ebdd02d93d6b2a7083debc55f28c9#r59003263>